sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapi/AADResourceServerOboConfigurationTest.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.aad.webapi;
+
+import org.junit.Test;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AADResourceServerOboConfigurationTest {
+
+    private static final String AAD_PROPERTY_PREFIX = "azure.activedirectory.";
+
+    private final WebApplicationContextRunner contextRunner = new WebApplicationContextRunner()
+        .withPropertyValues(
+            AAD_PROPERTY_PREFIX + "tenant-id=fake-tenant-id",
+            AAD_PROPERTY_PREFIX + "client-id=fake-client-id",
+            AAD_PROPERTY_PREFIX + "client-secret=fake-client-secret");
+
+    @Test
+    public void testNotExistBearerTokenAuthenticationToken() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testNotExistOAuth2LoginAuthenticationFilter() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(OAuth2LoginAuthenticationFilter.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testOnlyGraphClient() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withPropertyValues(AAD_PROPERTY_PREFIX + "authorization.graph.scopes=User.read")
```

Comment:
1. `User.read` -> `User.Read`. `R` should be upper case.
2. `scope` value should have prefix. i.e. it should be `http://xxx/User.Read`
3. It's better use more than one scope to test. Like `xxx.scopes=xxx/User.Read, xxx/User.Write`.

Reply:
Why is it better to use more than one scope to test?

Reply:
@saragluna  So we can test whether all scopes take effect.

Reply:
But how do we achieve that in the unit test?